src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -105,6 +105,25 @@ def __init__(self, storage=None, auth_ctx_factory=None, use_global_creds_cache=T
         self._management_resource_uri = CLOUD.endpoints.management
         self._ad_resource_uri = CLOUD.endpoints.active_directory_resource_id
 
+    def refresh_subscriptions(self):
+        # Currently requires a default subscription selected
+        existingSubscriptions = self.load_cached_subscriptions()
+        tenants = {}
+        bearer = self.get_raw_token()
+        self._creds_cache.flush_to_disk()
+        for sub in existingSubscriptions:
+            athctx = _authentication_context_factory(sub['tenantId'], self._creds_cache.adal_token_cache)
+            tenants.setdefault(sub['tenantId'], dict()).update({
+                'userId': sub['user']['name'],
+                "token": bearer[0][1],
+                "finder": SubscriptionFinder(athctx, self._creds_cache.adal_token_cache)})
+
+        for tenant in tenants:
```

Comment:
We should support new or removed tenants. If you decide not to do it, please open an issue